posts/tests.py

```diff
@@ -0,0 +1,312 @@
+@pytest.mark.django_db()
+def test_delete_poll_aouthor(new_poll):
+    user = new_poll.author
+    user.save()
+    new_poll.save()
+    assert new_poll in Poll.objects.all()
+    assert user in User.objects.all()
+    new_poll.delete()
+    assert new_poll not in Poll.objects.all()
+    assert user in User.objects.all()
+
+
+# Check if Poll's author deletion delete both the author and and the poll.
+@pytest.mark.django_db()
+def test_delete_poll(new_poll):
+    user = new_poll.author
+    user.save()
+    new_poll.save()
+    assert new_poll in Poll.objects.all()
+    assert user in User.objects.all()
+    user.delete()
+    assert new_poll not in Poll.objects.all()
+    assert user not in User.objects.all()
+
+
+# verify that it is impossible to create a Poll with unvalid params.
+def test_poll_unvalid_args():
+    with pytest.raises(TypeError):
+        Poll(user=USERNAME, description=DESCRIPTION)
+
+
+# -------------------------------------------- PollFile tests --------------------------------------------
+
+
+# Check that the PollFile values are the same as the PollFile inputs.
+def test_new_pollfile(new_pollFile):
+    assert new_pollFile.poll.author.username == USERNAME
+    assert new_pollFile.poll.description == DESCRIPTION
+    assert new_pollFile.poll.amount_of_voters == 0
+    assert new_pollFile.file == FILE1
+
+
+# Check if the PollFile is saved in the database and accessible via his Poll.
+@pytest.mark.django_db()
+def test_persist_pollfile(new_pollFile):
+    assert new_pollFile not in PollFile.objects.all()
+    new_pollFile.poll.author.save()
+    new_pollFile.poll.save()
+    new_pollFile.save()
+    assert new_pollFile in PollFile.objects.all()
+    assert new_pollFile in new_pollFile.poll.pollfile_set.all()
+
+
+# Check if PollFile deletion delete only PollFile from database.
+@pytest.mark.django_db()
+def test_delete_pollfile(new_pollFile):
+    user = new_pollFile.poll.author
+    poll = new_pollFile.poll
+    file = new_pollFile
+    user.save()
+    poll.save()
+    file.save()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert file in PollFile.objects.all()
+    file.delete()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert file not in PollFile.objects.all()
+
+
+# Check if PollFile's poll deletion delete both PollFile and poll from database - but not the poll's author.
+@pytest.mark.django_db()
+def test_delete_pollfile_poll(new_pollFile):
+    user = new_pollFile.poll.author
+    poll = new_pollFile.poll
+    file = new_pollFile
+    user.save()
+    poll.save()
+    file.save()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert file in PollFile.objects.all()
+    poll.delete()
+    assert poll not in Poll.objects.all()
+    assert user in User.objects.all()
+    assert file not in PollFile.objects.all()
+
+
+# Check if PollFile's poll's author deletion delete also PollFile and database.
```

Comment:
Same here: "...PollFIle's poll author..." instead of: "...PollFile's poll's author...".